common/persistence/historyV2Store.go

```diff
@@ -291,6 +291,9 @@ func (m *historyV2ManagerImpl) readHistoryBranch(byBatch bool, request *ReadHist
 	}
 
 	resp, err := m.persistence.ReadHistoryBranch(req)
+	if len(resp.History) == 0 && len(request.NextPageToken) == 0 {
+		err = &workflow.EntityNotExistsError{Message: "Workflow execution history not found."}
```

Comment:
I prefer to always return as soon as you can that way the reader knows they are done reading. But its a matter of preference.

Reply:
Updated.